packages/next/src/build/index.ts

```diff
@@ -1448,14 +1448,25 @@ export default async function build(
           )
         }
 
-        promises.push(
+        jobs.push(() =>
           handlePagesErrorRoute({
             currentIssues,
             entrypoints: currentEntrypoints,
             manifestLoader,
             rewrites: emptyRewritesObjToBeImplemented,
           })
         )
+        const promises = []
+        let i = 0
+        function nextJob(): Promise<void> | undefined {
+          if (i < jobs.length) {
+            return jobs[i++]().then(nextJob)
+          }
+        }
+        // Run 10 concurrently
+        for (; i < 10; i++) {
```

Comment:
You can probably use `next/dist/compiled/async-sema` for this instead: https://npmjs.com/async-sema